kuma/javascript/src/accountsettings/components/close-account.jsx

```diff
@@ -1,13 +1,16 @@
 // @flow
 import * as React from 'react';
-import { useState } from 'react';
 
+import { type UserData } from '../../user-provider.jsx';
 import { gettext } from '../../l10n.js';
 
-import CloseAccountForm from './close-account-form.jsx';
+type Props = {
+    locale: string,
+    userData: UserData,
+};
 
-const CloseAccount = () => {
-    const [showForm, setShowForm] = useState<boolean>(false);
+const CloseAccount = ({ locale, userData }: Props) => {
+    const { username }: any = userData;
```

Comment:
I don't understand this. Also, you can put this into the line above.

Reply:
I appreciate that it's a bit weird that type `UserData` makes the `username` optional. It makes sense and we should fix that, in Yari, so that you have an interface for authenticated and not authenticated users. E.g.
```typescript
interface BasicUser {
  isAuthenticated: boolean,
  waffle: ...,
  ...
}
interface AuthenticatedUser extends BasicUser {
  username: string;
  email: string; 
}
```

Reply:
The above was for that exact reason. Flow does not like the fact that `username` can be null or undefined and so, does not allow me to coerce it into a string